### PR TITLE
Create RT validations index and daily facts, and implement a single check on top of notices

### PR DESCRIPTION
**Pull request by atvaccaro**
# Description

Follow-up to https://github.com/cal-itp/data-infra/pull/1945, progress towards https://github.com/cal-itp/data-infra/issues/1904. Creates the infrastructure necessary to develop feed-level checks for RT up through `fct_daily_rt_feed_guideline_checks` with one example check implemented.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] agencies.yml

## How has this been tested?
Still working on it locally, tests pass but unsure if the models are good yet.

```
➜  warehouse git:(v2-rt-validations-framework) poetry run dbt build -s +fct_daily_rt_feed_guideline_checks+
20:12:18  Running with dbt=1.2.1
...
20:15:05  Finished running 19 view models, 103 tests, 1 seed, 6 table models, 3 incremental models in 0 hours 2 minutes and 38.91 seconds (158.91s).
20:15:05
20:15:05  Completed with 3 warnings:
20:15:05
20:15:05  Warning in test relationships_int_transit_database__urls_to_gtfs_datasets_gtfs_dataset_key__key__ref_dim_gtfs_datasets_ (models/intermediate/transit_database/_int_transit_database.yml)
20:15:05    Got 6 results, configured to warn if >0
20:15:05
20:15:05    compiled SQL at target/compiled/calitp_warehouse/models/intermediate/transit_database/_int_transit_database.yml/relationships_int_transit_data_15961df448f73cc529e2d786b56cd25f.sql
20:15:05
20:15:05  Warning in test relationships_fct_schedule_feed_downloads_gtfs_dataset_key__key__ref_dim_gtfs_datasets_ (models/mart/gtfs/_mart_gtfs.yml)
20:15:05    Got 20 results, configured to warn if != 0
20:15:05
20:15:05    compiled SQL at target/compiled/calitp_warehouse/models/mart/gtfs/_mart_gtfs.yml/relationships_fct_schedule_fee_f70034b868db33c646d494e151175683.sql
20:15:05
20:15:05  Warning in test relationships_fct_daily_schedule_feeds_gtfs_dataset_key__key__ref_dim_gtfs_datasets_ (models/mart/gtfs/_mart_gtfs.yml)
20:15:05    Got 15 results, configured to warn if != 0
20:15:05
20:15:05    compiled SQL at target/compiled/calitp_warehouse/models/mart/gtfs/_mart_gtfs.yml/relationships_fct_daily_schedu_039cc2c075c6cdbc7b9cf0c141dbf9c8.sql
20:15:05
20:15:05  Done. PASS=129 WARN=3 ERROR=0 SKIP=0 TOTAL=132
```

### CODE REVIEW

File: warehouse/models/mart/gtfs_quality/fct_daily_rt_feed_guideline_checks.md
Comment (lauriemerrell) [inline warehouse/models/mart/gtfs_quality/fct_daily_rt_feed_guideline_checks.md:13]: `base64_url` differs by feed type, so this is not aggregating across feed types (I think "any RT feed" is wrong here).

File: warehouse/models/mart/gtfs_quality/fct_daily_rt_feed_validation_notices.sql
Comment (lauriemerrell) [inline warehouse/models/mart/gtfs_quality/fct_daily_rt_feed_validation_notices.sql:28]: I feel like this whole CTE could maybe use a few comments or something, this was incredibly hard for me to parse especially because it's aggregating across a bunch of different grains at once

File: warehouse/seeds/gtfs_rt_validation_code_descriptions.csv
Comment (lauriemerrell) [inline warehouse/seeds/gtfs_rt_validation_code_descriptions.csv:1]: is it possible to comment where this came from, especially version information?

File: warehouse/seeds/gtfs_schedule_validation_code_descriptions.csv
Comment (lauriemerrell) [inline warehouse/seeds/gtfs_schedule_validation_code_descriptions.csv:1]: same Q: is it possible to comment where this came from, especially version information?

File: warehouse/dbt_project.yml
Comment (lauriemerrell) [inline warehouse/dbt_project.yml:25]: did you check if there are any other incremental RT tables that should use this?

File: warehouse/models/mart/gtfs_quality/_mart_gtfs_quality.yml
Comment (lauriemerrell) [reply warehouse/models/mart/gtfs_quality/_mart_gtfs_quality.yml:23]: The merge on this is gonna be a little gross, especially with you changing the name of the schedule feed check table

**Review by lauriemerrell — CHANGES_REQUESTED** [review]
mostly just doc requests 🙏

File: warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__rt_validation_notices.sql
Comment (atvaccaro) [reply warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__rt_validation_notices.sql:4]: Oh it's the default https://docs.getdbt.com/reference/resource-configs/bigquery-configs#partition-clause:~:text=Daily%20partitioning%20is%20the%20default%20for%20all%20column%20types. I can make it explicit for clarity

File: warehouse/models/mart/gtfs_quality/fct_daily_rt_feed_guideline_checks.md
Comment (atvaccaro) [reply warehouse/models/mart/gtfs_quality/fct_daily_rt_feed_guideline_checks.md:13]: Ah yeah let me clarify since this isn't rolled up to the quartet yet.

File: warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__no_rt_critical_validation_errors.sql
Comment (lauriemerrell) [reply warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__no_rt_critical_validation_errors.sql:42]: Ok, still would want to confirm with him -- not sure if we want to carry that behavior forward.

File: warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__rt_validation_notices.sql
Comment (lauriemerrell) [reply warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__rt_validation_notices.sql:4]: ok I was mostly just wondering why it was specified in one but not the other

File: warehouse/seeds/gtfs_rt_validation_code_descriptions.csv
Comment (atvaccaro) [reply warehouse/seeds/gtfs_rt_validation_code_descriptions.csv:1]: yep will document/test seeds

File: warehouse/dbt_project.yml
Comment (atvaccaro) [reply warehouse/dbt_project.yml:25]: I don't think so right now, because we don't actually materialize actual RT data right now.

File: warehouse/models/mart/gtfs_quality/_mart_gtfs_quality.yml
Comment (atvaccaro) [reply warehouse/models/mart/gtfs_quality/_mart_gtfs_quality.yml:23]: Yep I will rebase next week and ensure everything is smooth for @owades

File: warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__rt_validation_notices.sql
Comment (atvaccaro) [reply warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__rt_validation_notices.sql]: Just inconsistency on my part :(

**Comment by atvaccaro** [comment]
Planning on waiting until I am back next week to merge this, so I can coordinate with @owades to rebase on top of his work once it is merged.

File: warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__no_rt_critical_validation_errors.sql
Comment (owades) [reply warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__no_rt_critical_validation_errors.sql:43]: What she said! When we don't have validator data for a given day let's leave null.
A potential addition would be to create a "reason" column as well, which could add some kind of detail. It would be smart for me to create this now...

**Review by owades — APPROVED** [review]
Looks good. Love the validation outcome descriptions.

**Comment by lauriemerrell** [comment]
Maybe can fix this here: https://sentry.calitp.org/organizations/sentry/issues/13528/?project=2&referrer=slack

File: warehouse/models/mart/gtfs_quality/_mart_gtfs_quality.yml
Comment (atvaccaro) [reply warehouse/models/mart/gtfs_quality/_mart_gtfs_quality.yml:27]: > The merge on this is gonna be a little gross, especially with you changing the name of the schedule feed check table

This didn't end up being too bad.

File: warehouse/models/mart/gtfs_quality/fct_daily_rt_feed_validation_notices.sql
Comment (lauriemerrell) [inline warehouse/models/mart/gtfs_quality/fct_daily_rt_feed_validation_notices.sql:48]: just confirming that we will probably need to fix this too in #1994 (less urgent than schedule because schedule we've already changed the version)

**Review by lauriemerrell — APPROVED** [review]
Just one comment, I think this looks good, we can see how it plays out once we reindex with quartets too

**Pull request merged**
